added Friends page (/friends)

paramjeet-dev opened the pull request:
## Description
On the sessions page, if the suggested friends are more than 15, only first 15 users are shown and a link to open a new page (/friends) is shown. On clicking the link, the user is redirected to /friends where he/she can see all suggested friends, incoming friends requests, sent requests and all friends. they can also manage those requests from the page.

## Related Issue
Fixes #233 

## Changes Made
- added new folder friendsPage in components section that handles various UI components for the page.
- updated Session.jsx to add a condition to handle >15 suggested friends.
-  added friends page.jsx file to show the /friends page to users.
- updated friendController.js to handle cancel sent requests.

## Screenshots or GIFs (if applicable)
<img width="1919" height="945" alt="Screenshot 2025-08-10 133529" src="https://github.com/user-attachments/assets/df9094e5-11d3-42f3-a0b5-480bb223b43f" />
<img width="1917" height="949" alt="Screenshot 2025-08-10 133507" src="https://github.com/user-attachments/assets/5be84622-cd7f-4c85-95f6-9477e77db97d" />
<img width="1917" height="933" alt="Screenshot 2025-08-10 133448" src="https://github.com/user-attachments/assets/c8bf0e1e-4996-48cc-8879-7c68e410fb88" />
<img width="1917" height="941" alt="Screenshot 2025-08-10 133435" src="https://github.com/user-attachments/assets/df6abff3-0e39-4bf2-9b14-02d90127961e" />

the below image is a sample to show the link that redirects the user. In functionality, this will be visible only when suggested friends are more than 15. Added this ss as reference only.:
<img width="411" height="304" alt="Screenshot 2025-08-10 133624" src="https://github.com/user-attachments/assets/300d1b33-6ac5-4d59-8c33-7e33dbd63802" />
 
## Checklist
- [✅ ] I have performed a self-review of my code.
- [✅ ] My changes are well-documented.
- [✅ ] I have added tests that prove my fix is effective or that my feature works.
- [✅ ] Any dependent changes have been merged and published.

RishuKumarCodes commented:
@paramjeet-dev if it already shows name image profile, you can add field of study ( if user have added) that will help them fining people with same interest. no need to add anything else.
- Hoping to see impressive changes this time.

paramjeet-dev commented:
@RishuKumarCodes 
<img width="1347" height="394" alt="Screenshot 2025-08-10 212749" src="https://github.com/user-attachments/assets/e997eafb-1bc9-44eb-873e-4c86b8062fa3" />

- suggested friends tab example attached.
- added a div to show user intrests
- fixed div of user card to show 2 cards (pc) and 1 card (mobile) per row

RishuKumarCodes commented:
@paramjeet-dev your changes look good. Thanks for your PR.
